Back-To-School/Assets/BackToSchool/Scripts/GameManagement/Game.cs

```diff
@@ -0,0 +1,179 @@
+        private StatsManager _statsManager;
+        private SaveSystem _saveSystem;
+
+        private float _gameOverDelay = 1f;
+        private bool _isGamePaused;
+        private bool _isPlayerDead;
+
+        public void Initialize(SaveSystem saveSystem, GameManager gameManager, GameParameters parameters)
+        {
+            _saveSystem   = saveSystem;
+            _gameManager  = gameManager;
+            _playerStats  = new PlayerStats();
+            _levelSystem  = new LevelSystem();
+            _statsManager = new StatsManager();
+
+            _player       = Instantiate(_playerPrefab, transform.position, Quaternion.identity);
+            _inputManager = Instantiate(_inputManagerPrefab, transform.position, Quaternion.identity).GetComponent<InputManager>();
+            _enemySpawner = Instantiate(_enemySpawnerPrefab, transform.position, Quaternion.identity).GetComponent<EnemySpawner>();
+
+            SubscribeEvents();
+            _statsManager.Initialize(_playerStats, parameters.InitialLevel);
+
+            _mainCamera.GetComponent<CameraFollow>().SetTarget(_player.transform);
+            _enemySpawner.SetTarget(_player.gameObject);
+
+            _playerInput = new PlayerInputProvider(_mainCamera);
+            _inputManager.Subscribe(_playerInput);
+            _player.Initialize(_playerInput, _playerStats);
+
+            var pauseInput = new PauseInputProvider();
+            _inputManager.Subscribe(pauseInput);
+            pauseInput.Cancelled += OnGameStopped;
+
+            if (parameters.IsNewGame)
+            {
+                _saveSystem.ResetPlayerProgress();
+                _player.InitializeHealth();
+                _player.WeaponController.InitializeWeapon();
+            }
+            else
+                _saveSystem.LoadPlayerProgress(_player, _levelSystem);
+        }
+
+        private void SubscribeEvents()
+        {
+            _gameOverPresenter.Restarted += RestartGame;
+            _pausePresenter.Restarted    += RestartGame;
+            _pausePresenter.Continued    += ContinueGame;
+            _pausePresenter.MenuReturned += ReturnToMenu;
+
+            _player.WeaponController.AmmoChanged    += _hudPresenter.OnAmmoChanged;
+            _player.WeaponController.WeaponChanged  += _hudPresenter.OnWeaponChanged;
+            _player.WeaponController.MaxAmmoChanged += _hudPresenter.OnMaxAmmoChanged;
+            _player.Died                            += OnPlayerDeath;
+            _player.HealthChanged                   += _hudPresenter.OnHealthChanged;
+            _enemySpawner.EnemyDied                 += _levelSystem.AddExperience;
+
+            _levelSystem.LevelChanged      += _statsManager.LevelUp;
+            _levelSystem.LevelChanged      += _hudPresenter.OnLevelChanged;
+            _levelSystem.ExperienceChanged += _hudPresenter.OnExpChanged;
+            _saveAction                    =  delegate { _saveSystem.SavePlayerProgress(_player, _levelSystem); };
```

Comment:
There is no need to such complicated delegate. Just make a method:
```
        private void SaveGame()
        {
            _saveSystem.SavePlayerProgress(_player, _levelSystem);
        }
```
and subscribe to it like:
```
_levelSystem.ProgressChanged   += SaveGame;
```
And after that you can fully remove _saveAction